DuckDuckGoTests/Migration/MigrationTests.swift

```diff
@@ -0,0 +1,225 @@
+        let _ = createStory(in: feed)
+        XCTAssert(container.save())
+        
+        let expectation = XCTestExpectation(description: "testMigrateBothTypes")
+        Migration(container: container).start { storiesMigrated, bookmarksMigrated in
+            XCTAssertEqual(2, storiesMigrated)
+            XCTAssertEqual(2, bookmarksMigrated)
+            
+            let bookmarksManager = BookmarksManager()
+            XCTAssertEqual(4, bookmarksManager.count)
+            
+            expectation.fulfill()
+        }
+        
+        wait(for: [expectation], timeout: 1)
+    }
+    
+    func testOldBookmarksDeletedAfterMigration() {
+        testSingleFavouriteSearchesMigratedToBookmarks()
+        XCTAssertNil(UserDefaults.standard.array(forKey: Migration.oldBookmarksKey))
+    }
+    
+    func testSeveralFavouriteSearchesMigratedToBookmarks() {
+        
+        let defaults = UserDefaults.standard
+        defaults.setValue([[ "title": "example1.com", "url": "http://www.example1.com" ],
+                           [ "title": "example2.com", "url": "http://www.example2.com" ],
+                           [ "title": "example3.com", "url": "http://www.example3.com" ]], forKeyPath: Migration.oldBookmarksKey)
+        defaults.synchronize()
+        
+        let expectation = XCTestExpectation(description: "testSeveralFavouriteSearchesMigratedToBookmarks")
+        Migration(container: container).start { storiesMigrated, bookmarksMigrated in
+            XCTAssertEqual(0, storiesMigrated)
+            XCTAssertEqual(3, bookmarksMigrated)
+            
+            let bookmarksManager = BookmarksManager()
+            XCTAssertEqual(3, bookmarksManager.count)
+            
+            expectation.fulfill()
+        }
+        
+        wait(for: [expectation], timeout: 1)            
+    }
+
+    func testSingleFavouriteSearchesMigratedToBookmarks() {
+        
+        let defaults = UserDefaults.standard
+        defaults.setValue([[ "title": "example.com", "url": "http://www.example.com" ]], forKeyPath: Migration.oldBookmarksKey)
+        defaults.synchronize()
+        
+        let expectation = XCTestExpectation(description: "testSingleFavouriteSearchesMigratedToBookmarks")
+        Migration(container: container).start { storiesMigrated, bookmarksMigrated in
+            XCTAssertEqual(0, storiesMigrated)
+            XCTAssertEqual(1, bookmarksMigrated)
+            
+            let bookmarksManager = BookmarksManager()
+            XCTAssertEqual(1, bookmarksManager.count)
+            
+            let link = bookmarksManager.bookmark(atIndex: 0)
+            XCTAssertEqual("example.com", link.title)
+            XCTAssertEqual("http://www.example.com", link.url.absoluteString)
+            XCTAssertNil(link.favicon)
+            
+            expectation.fulfill()
+        }
+        
+        wait(for: [expectation], timeout: 1)
+    }
+    
+    func testFavouriteStoriesDeletedAfterMigration() {
```

Comment:
It now clears existing book marks as part of the test setup - hopefully safe to assume people with bookmarks won't be running unit tests against their actual boomarks. :)